waku/waku_rln_relay/rln_relay.nim

```diff
@@ -308,21 +308,41 @@ proc appendRLNProof*(
   let proof = rlnPeer.groupManager.generateProof(input, epoch, nonce).valueOr:
     return err("could not generate rln-v2 proof: " & $error)
 
-
   msg.proof = proof.encode().buffer
   return ok()
 
+proc compareKeys(a, b: Epoch): int =
+  for i in 0..<32:
+    if a[i] < b[i]:
+      return -1
+    elif a[i] > b[i]:
+      return 1
+  return 0
+
 proc clearNullifierLog*(rlnPeer: WakuRlnRelay) =
   # clear the first MaxEpochGap epochs of the nullifer log
   # if more than MaxEpochGap epochs are in the log
   # note: the epochs are ordered ascendingly
   if rlnPeer.nullifierLog.len().uint <= rlnPeer.rlnMaxEpochGap:
     return
 
+  echo "elements in nullifier log"
+  for i in rlnPeer.nullifierLog.keys().toSeq():
+    echo i
   let countToClear = rlnPeer.nullifierLog.len().uint - rlnPeer.rlnMaxEpochGap
   trace "clearing epochs from the nullifier log", count = countToClear
-  let epochsToClear = rlnPeer.nullifierLog.keys().toSeq()[0 ..< countToClear]
+
+  #let epochsToClear = rlnPeer.nullifierLog.keys().toSeq()[0 ..< countToClear]
+  #let epochsToClear = rlnPeer.nullifierLog.keys().toSeq().sort(compareKeys)[0 ..< countToClear]
+
+  var epochsToClear = rlnPeer.nullifierLog.keys().toSeq()
+  epochsToClear.sort(compareKeys)
+  epochsToClear = epochsToClear[0 ..< countToClear]
```

Comment:
as it is, it should work fine now.

however I'm having second thought about cleaning the nullifierlog this way. it has some implicit knowledge that can be error prone.

perhaps a better way could be. which epoch are we at? x? well, remove everything in the table that is beyond x-epochGap.

Reply:
i like this idea